probe-rs/src/probe/jlink/mod.rs

```diff
@@ -941,6 +951,31 @@ impl DAPAccess for JLink {
     }
 }
 
+impl SwoAccess for JLink {
+    fn enable_swo(&mut self, config: &SwoConfig) -> Result<(), ProbeRsError> {
+        // TODO: Return ProbeRsErrors
+        let jlink = self.handle.get_mut().unwrap();
+        jlink
+            .swo_start_uart(config.baud(), SWO_BUFFER_SIZE.into())
+            .unwrap();
+        Ok(())
+    }
+
+    fn disable_swo(&mut self) -> Result<(), ProbeRsError> {
+        // TODO: Return ProbeRsErrors
+        let jlink = self.handle.get_mut().unwrap();
+        jlink.swo_stop().unwrap();
+        Ok(())
+    }
+
+    fn read_swo_timeout(&mut self, _timeout: std::time::Duration) -> Result<Vec<u8>, ProbeRsError> {
+        // TODO: Return ProbeRsErrors
+        // TODO: consider timeout
+        let data = self.read_swo_data()?;
```

Comment:
This should really timeout. The problem here is that people expect this to not be blocking forever. If it is not currently possible to timeout, I would rather have this return an error. But ultimately this should really timeout.

Reply:
`read_swo_data` does a timeout and that one relies on the default timeout here: https://docs.rs/jaylink/0.1.4/src/jaylink/lib.rs.html#466

If you are ok with it, I would re-add the `read_swo` without timeout and let the `read_swo_timeout` do `unimplemented!()`

Reply:
It looks like that jaylink function actually blocks until the entire buffer is filled with SWO data, i.e. it will never return early or only return the currently-available data. Perhaps jaylink could gain a function to only do a single `read_bulk` operation with a specified timeout, which would match the probe-rs semantics?